agent/pkg/k8s/certificates.go

```diff
@@ -80,13 +82,31 @@ func (c *Client) getControlPlaneComponentPod(component string) (*v1.Pod, error)
 func getServerName(podsa string, podns string, container *v1.Container) (string, error) {
 	var l5dns string
 	var l5dtrustdomain string
+	var localName string
+
 	for _, env := range container.Env {
 		if env.Name == linkerdNsEnvVarName {
 			l5dns = env.Value
 		}
 		if env.Name == linkerdTrustDomainEnvVarName {
 			l5dtrustdomain = env.Value
 		}
+
+		if env.Name == linkerdProxyIdentityEnvVarName {
+			localName = env.Value
+		}
+	}
+
+	if l5dns == "" && l5dtrustdomain == "" {
+		// on newer versions of Linkerd these are not set, rely on the proxy local name for this
```

Comment:
can you describe a bit more in comments the difference between the linkerd versions? also, specify which version this changed?

something like...

old / `stable-2.8.1`:
```yaml
_l5d_ns: linkerd
_l5d_trustdomain: cluster.local
LINKERD2_PROXY_IDENTITY_LOCAL_NAME: $(_pod_sa).$(_pod_ns).serviceaccount.identity.$(_l5d_ns).$(_l5d_trustdomain)
```
new / `edge-22.2.2`:
```yaml
LINKERD2_PROXY_IDENTITY_LOCAL_NAME: $(_pod_sa).$(_pod_ns).serviceaccount.identity.linkerd.cluster.local
```